packages/core/src/components/select/select.tsx

```diff
@@ -48,10 +50,80 @@ export class AtomSelect {
   @Event() atomDismiss!: EventEmitter<void>
   @Event() atomFocus!: EventEmitter<void>
 
+  @Method()
+  setTag() {
+    /**
+     * This method was necessary because the `ion-selection-option` loop does not allow customizations or custom components.
+     * So, to be able to add custom elements such as a tag or a badge inside an option of the `select` field, when the select
+     * is opened, the `onBlur` event triggers this method that performs a search for all `ion-item` elements (which is the
+     * final element rendered to list options) and filters the ones that need to be changed.
+     */
+
+    const ionItemElements = document.querySelectorAll('ion-item') || []
+
+    ionItemElements?.forEach((itemElement) => {
+      const optionText = itemElement.textContent?.trim()
+      const optionWithTag = this.optionsWithTag[optionText]
+
+      if (!optionWithTag) return
+
+      const { color, label } = optionWithTag.tag
+
+      const optionElement =
+        this.getElementByTag(itemElement, 'ion-radio') ||
+        this.getElementByTag(itemElement, 'ion-checkbox')
+      const optionShadowRoot = optionElement.shadowRoot
+        .firstElementChild as HTMLElement
+      const firstElementInOption =
+        optionShadowRoot.firstElementChild as HTMLElement
+
+      const tagElement = document.createElement('atom-tag')
+
+      tagElement.setAttribute('color', color)
+      tagElement.style.marginLeft = '8px'
```

Comment:
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches and the blocking or non-blocking decorations to further communicate your intent.  -->
**suggestion:** ‏we could use ```var(--spacing-xsmall)```

Reply:
https://github.com/juntossomosmais/atomium/pull/549#:~:text=fix%3A%20pr%20tips-,beced49,-Merge%20state